src/node_dotenv.cc

```diff
@@ -98,12 +106,18 @@ bool Dotenv::ParsePath(const std::string_view path) {
     result.append(buf.base, r);
   }
 
-  using std::string_view_literals::operator""sv;
-  auto lines = SplitString(result, "\n"sv);
 
-  for (const auto& line : lines) {
-    ParseLine(line);
+  EnvStream env_stream(&result);
+
+  std::vector<EnvPair*> env_pairs;
+  EnvReader::read_pairs(&env_stream, &env_pairs);
+
+  for (const auto pair : env_pairs) {
+    EnvReader::finalize_value(pair, &env_pairs);
+    store_.insert_or_assign(*pair->key->key, *pair->value->value);
   }
+  EnvReader::delete_pairs(&env_pairs);
```

Comment:
Why do we delete pairs?

Reply:
Because store is a map and map makes a copy of the it's values once the process is complete.